modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,371 @@
+import { afterEach, beforeEach, context, describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import * as Chai from 'chai';
+import { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+import type Editor from 'tinymce/core/api/Editor';
+import { createUserLookup, type User } from 'tinymce/core/lookup/UserLookup';
+
+Chai.use(chaiAsPromised);
+
+const createMockUser = (id: string): User => ({
+  id,
+  name: 'Test User',
+  avatar: 'test-avatar.png',
+  description: 'Test Description'
+});
+
+describe('browser.tinymce.core.UserLookupTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    user_id: 'test-user-1',
+    fetch_users: (userIds: string[]): Promise<User[]> => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const validIds = Arr.filter(userIds, (id): id is string =>
+            typeof id === 'string' && id.length > 0
+          );
+
+          // Only create users for valid test IDs
+          const users = Arr.bind(validIds, (id) =>
+            id.startsWith('test-user-') ? [ createMockUser(id) ] : []
+          );
+
+          resolve(users);
+        }, 0);
+      });
+    },
+  });
+
+  context('getUserId', () => {
+    it('TINY-11974: Should return the configured user ID', () => {
+      const editor = hook.editor();
+      const currentUserId = editor.userLookup.getUserId();
+      expect(currentUserId).to.equal('test-user-1', 'Should return the configured user ID');
+    });
+
+    it('TINY-11974: Should return an immutable user ID value', () => {
+      const editor = hook.editor();
+      const originalUserId = 'test-user-1';
+      const userId = editor.userLookup.getUserId();
+
+      expect(() => {
+        Object.defineProperty(userId, '0', {
+          value: 'n'
+        });
+      }).to.throw();
+
+      expect(userId).to.equal(originalUserId);
+    });
+  });
+
+  context('fetchUsers', () => {
+    it('TINY-11974: Should fetch users and return array of promises', () => {
+      const editor = hook.editor();
+      const userIds = [ 'test-user-1' ];
+
+      const promises = editor.userLookup.fetchUsers(userIds);
+
+      expect(promises).to.be.an('array', 'Should return an array of promises');
```

Comment:
Do we have to use this TDD style asserts we don't use them anywhere else and I never understood the benefit of them to be honest. Reminds me of languages from past where they tried to use plain English as the language. 😄

Reply:
I thought we had more, looks like it's only in some premium repo tooling.

When I asked Kim to switch to chai I didn't really want a conversion to use `expect` everywhere, but I really want that easy "is rejected with" assertion which is only available in that style. So we might as well be consistent (at least in this file, I don't think we should expand it).